File: Sheet 6 (Arrays 1)/K.cpp
#include <iostream>
#include <algorithm>
using namespace std;
int main()
{
    int n , m , arr[55];
    cin >> n >> m ;
    for (int i = 0 ; i <  m ; i ++){
        cin >> arr[i];
    }
    sort(arr ,arr+m);
    int mini = 10065 ;
    for (int i = 0 ; i < m ; i ++){
        if (i+n-1 >= m)break;
        if (mini > arr[i+n-1]- arr[i])mini = arr[i+n-1]- arr[i];
    }
    cout << mini ;
return 0;
}

File: Sheet 6 (Arrays 1)/K.py
n , m = input().split()
n = int(n) ; m = int(m)
a = input()
l = a.split()
for i in range(len(l)):
    l[i]= int(l[i])
l.sort()
mini =10123123123
for i in range(len(l) - n + 1):
    mini = min(mini , l[i+n-1] - l[i])
print(mini)
